### ISSUE TITLE
Suggestion: start_active_span should return the block return

### ISSUE THREAD

**Issue by ethanjcohen-underarmour** [opentracing/opentracing-ruby#41]
Has anyone discussed having start_active_span return the block return, rather than the scope?
https://github.com/opentracing/opentracing-ruby/blob/master/lib/opentracing/tracer.rb#L48

Returning the scope makes it really hard to wrap existing code without refactoring.

For example, if I have this simple function:
```
def get_name
    "ethan"
end
```

and I want to trace it like this...
```
def get_name
  OpenTracing.start_active_span('get_name') do
    "ethan"
  end
end
```

then this happens
```
puts get_name -> #<LightStep::Scope:0x00557203673458>
```

Using the not-block version of the function doesn't help much either, I'd still have to do something like this:
```
def get_name
  scope = OpenTracing.start_active_span('get_name')
  r = "ethan"
  scope.close

  r
end
```

I realize this suggestion is a big breaking change, but this would allow for far easier integration with existing code (especially complex functions with multiple returns) and easier chaining. Thoughts?

**Comment by austinlparker** [comment]
cc @mwear

**Comment by mwear** [comment]
Thanks for this suggestion @ethanjcohen-underarmour. 

I can see having `Tracer#start_active_span` return the value of the block rather than the scope making this API more ergonomic. Since the scope is yielded to block, it would be easy to return it from the block if that was what was desired.

Given that this would be a breaking change, I'd like to get input from tracer implementers before moving forward with it.

cc: @indrekj @palazzem @austinlparker

**Comment by indrekj** [comment]
It does make sense. You could do
```
def get_name
  name = nil
  OpenTracing.start_active_span('get_name') do
    name = "ethan"
  end
  name
end
```
but that's just ugly. The scope will be closed anyway when the block ends so there's really no need to access the scope after the block has been executed.

I went over instrumentation libraries that we're using
* https://github.com/opentracing-contrib/ruby-rack-tracer
* https://github.com/opentracing-contrib/ruby-faraday-tracer/
* https://github.com/salemove/ruby-activerecord-opentracing
* https://github.com/salemove/ruby-sidekiq-tracer

These libraries and our code would be compatible with this change.

I think this is a reasonable request. So the question is how to rollout this with minimal impact. I'm willing to change jaeger and zipkin tracers, but I don't know if somebody is relying on the current behaviour.

**Comment by ethanjcohen-underarmour** [comment]
I suspect you'd have little to no impact on existing code if you changed it to return the value from the block only when a block is provided, otherwise return the scope/span like normal.

I generally _try_ not to return multiple types from functions, but it _feels_ like it would be ok in this case. Are there any precedents of that in ruby?

**Comment by austinlparker** [comment]
I’m not much of a Rubyist but this seems sound. I’d be most in favor of the optional block idea - if we can do this without breaking the existing API, so much the better.

**Comment by mwear** [comment]
Addressed in #45 and released in [opentracing 0.5.0](https://rubygems.org/gems/opentracing/versions/0.5.0).
